### PR TITLE
Fix vrange min/max adjusting when min exceeds max

### PR DESCRIPTION
**Pull request by kilbith**
Title says it all... @est31

### CODE REVIEW

**Comment by nerzhul** [comment]
@paramat FIX ANDROID BUILD, please :) thanks you :p

**Comment by paramat** [comment]
@nerzhul see https://github.com/minetest/minetest/pull/3088#issuecomment-145985408

**Comment by paramat** [comment]
Fixed, now you can rebase this commit to pass checks.

**Comment by est31** [comment]
Let's keep this PR open, perhaps there are more things that need fixing.

File: builtin/mainmenu/tab_settings.lua [hunk 1]
```diff
@@ -329,10 +329,6 @@ local function formspec(tabview, name, tabdata)
 		 		viewing_range_max_to_scrollbar() .. "]" ..
 		 	"tooltip[sb_viewing_range_max;" .. fgettext("Maximal Viewing Range: $1 nodes",
 				get_vrange_max) .. "]"
-
-		if (get_vrange_min == nil and get_vrange_min > get_vrange_max) then
-			core.setting_set("viewing_range_nodes_min", get_vrange_max)
-		end
 	elseif (core.setting_getbool("viewing_range_auto") == nil or
 			core.setting_getbool("viewing_range_auto") == false) then
```
Comment (kwolekr) [inline builtin/mainmenu/tab_settings.lua]: Why is this elseif statement not just an else?

File: builtin/mainmenu/tab_settings.lua [hunk 2]
```diff
@@ -346,6 +342,9 @@ local function formspec(tabview, name, tabdata)
 			"checkbox[3.85,4.3;cb_vrange_auto_adjust;" .. fgettext("Auto-adjust") .. ";" ..
 				dump(core.setting_getbool("viewing_range_auto")) .. "]"
 	end
+	if get_vrange_min > get_vrange_max then
+		core.setting_set("viewing_range_nodes_min", get_vrange_max)
```
Comment (kwolekr) [inline builtin/mainmenu/tab_settings.lua]: Why have this correction after the formspec has already been rendered?

File: builtin/mainmenu/tab_settings.lua [hunk 3]
```diff
@@ -346,6 +342,9 @@ local function formspec(tabview, name, tabdata)
 			"checkbox[3.85,4.3;cb_vrange_auto_adjust;" .. fgettext("Auto-adjust") .. ";" ..
 				dump(core.setting_getbool("viewing_range_auto")) .. "]"
 	end
+	if get_vrange_min > get_vrange_max then
```
Comment (kwolekr) [inline builtin/mainmenu/tab_settings.lua]: This would crash if either of these variables are nil.

Comment (est31) [inline builtin/mainmenu/tab_settings.lua]: How can they be nil?

File: builtin/mainmenu/tab_settings.lua (hunk 1 above)
Comment (kilbith) [inline builtin/mainmenu/tab_settings.lua]: Right, was too zealous for prevention here...

File: builtin/mainmenu/tab_settings.lua (hunk 3 above)
Comment (kilbith) [inline builtin/mainmenu/tab_settings.lua:344]: They can't : https://github.com/minetest/minetest/blob/master/src/defaultsettings.cpp#L95-L96

File: builtin/mainmenu/tab_settings.lua (hunk 2 above)
Comment (kilbith) [inline builtin/mainmenu/tab_settings.lua:345]: 1. This is always being rendered within the function `formspec()`, scrollbar is dynamically updated according to values and inversely.
2. This checking deals only with the setting values, thus that doesn't matter which string the formspec currently has (i.e. it behaves the same than in the if statement or everywhere else, just matter of style).

And no, this ain't a dupe of previous PR, the condition has changed.

File: builtin/mainmenu/tab_settings.lua (hunk 3 above)
Comment (kwolekr) [inline builtin/mainmenu/tab_settings.lua]: Okay, fair enough.  But what if a user accidentally sets it to a string value instead?

Comment (kilbith) [inline builtin/mainmenu/tab_settings.lua:344]: > But what if a user accidentally sets it to a string value instead?

This doesn't cause an error at least. If max is a string, min is only considered (and vice-versa). If both are string, the vrange is set to 15.

**Comment by kilbith** [comment]
Closing, will be obsoleted by https://github.com/minetest/minetest/pull/3258. Thanks for the review.

**Pull request closed without merging**